tests/integration-tests/host-exports/src/mapping.ts

```diff
@@ -63,3 +68,36 @@ export function handleTrigger(event: Trigger): void {
   // Test big int right shift
   assert(bigInt >> 6 == BigInt.fromString("138888888888888"));
 }
+
+function testEthereumAbi(): void {
+  let address = ethereum.Value.fromAddress(Address.fromString("0x0000000000000000000000000000000000000420"));
+  let bigInt1 = ethereum.Value.fromUnsignedBigInt(BigInt.fromI32(62));
+  let bigInt2 = ethereum.Value.fromUnsignedBigInt(BigInt.fromI32(63));
+  let bool = ethereum.Value.fromBoolean(true);
+
+  let fixedSizedArray = ethereum.Value.fromFixedSizedArray([
+    bigInt1,
+    bigInt2
+  ]);
+
+  let tupleArray: Array<ethereum.Value> = [
+    fixedSizedArray,
+    bool
+  ];
+
+  let tuple = ethereum.Value.fromTuple(tupleArray as ethereum.Tuple);
+
+  let params: Array<ethereum.Value> = [
+    address,
+    tuple
+  ];
+
+  let encoded = ethereum.encode(params)!;
+
+  let decoded = ethereum.decode("(address,(uint256[2],bool))", encoded).toTuple();
+
+  assert(address.toAddress() == decoded[0].toAddress(), "address ethereum encoded does not equal the decoded value");
+  assert(bigInt1.toBigInt() == decoded[1].toTuple()[0].toArray()[0].toBigInt(), "uint256[0] ethereum encoded does not equal the decoded value");
+  assert(bigInt2.toBigInt() == decoded[1].toTuple()[0].toArray()[1].toBigInt(), "uint256[1] ethereum encoded does not equal the decoded value");
+  assert(bool.toBoolean() == decoded[1].toTuple()[1].toBoolean(), "boolean ethereum encoded does not equal the decoded value");
```

Comment:
It would be good to use some intermediary variables to make the test more readable. It's great to have a test for tuples, could we also test for a simple type, and also for nested tuples? e.g. `(address,(address, bool))`.

Reply:
Done!